### PR TITLE
Update pangolin to 4.3.1

### PR DESCRIPTION
**Pull request by BiocondaBot**
Update [`pangolin`](https://bioconda.github.io/recipes/pangolin/README.html): **4.3** &rarr; **4.3.1**

[install with bioconda](http://bioconda.github.io/recipes/pangolin/README.html) [Conda](https://anaconda.org/bioconda/pangolin/files)

Info | Link or Description
-----|--------------------
Recipe | [`recipes/pangolin`](https://github.com//bioconda/bioconda-recipes/tree/bump/pangolin/recipes/pangolin) (click to view/edit other files)
Summary | Phylogenetic Assignment of Named Global Outbreak LINeages
Home | [https://github.com/cov-lineages/pangolin](https://github.com/cov-lineages/pangolin)
Releases |[https://github.com/cov-lineages/pangolin/tags](https://github.com/cov-lineages/pangolin/tags)
Author | `@cov-lineages`
***

This pull request was automatically generated (see [docs](https://bioconda.github.io/contributor/updating.html)).

### CODE REVIEW

**Comment by wm75** [comment]
Leaving this one open to prevent the bot from opening a new one, but see #42191 .

**Comment by corneliusroemer** [comment]
@wm75  I'm not quite sure why we you don't just want merge this - it doesn't matter if there are no code changes, why not have a bioconda package anyways? Some people may want to pull that version as they know it exists.

**Comment by wm75** [comment]
yeah, saw the pangolin issue so apparently that's a real thing.

No strong feelings about this from my side, I merely thought I'd stop propagation of bot PRs upstream (next thing that's going to happen is an autogenerated update PR for Galaxy where we do not want the extra version).
If you think it's useful to have the release in bioconda, just merge this PR and I'll close the Galaxy one once it appears.

File: recipes/pangolin/meta.yaml [hunk 1]
```diff
@@ -19,23 +19,23 @@ requirements:
     - pip
   run:
     - python >=3.7
-    - biopython ==1.74
+    - biopython >=1.74
     - pandas >=1.0.1
-    - tabulate
     - joblib >=0.11
-    - scikit-learn <1.3.0
+    - scikit-learn >=0.23.1
```
Comment (wm75) [inline recipes/pangolin/meta.yaml:25]: This one is not helpful, see https://github.com/bioconda/bioconda-recipes/pull/42145/commits/ce574e73f5061c4f32416c2d341cbfcfcfcb543b

File: recipes/pangolin/meta.yaml [hunk 2]
```diff
@@ -19,23 +19,23 @@ requirements:
...
     - pulp >=2
     - minimap2 >=2.16
-    - snakemake-minimal >=5.13,<=7.30.1
+    - snakemake-minimal >=5.13,!=7.30.1
     - gofasta
-    - usher >=0.6.2
+    - usher >=0.5.4
```
Comment (wm75) [inline recipes/pangolin/meta.yaml:30]: see https://github.com/cov-lineages/pangolin/issues/500 for why usher is pinned to >=0.6.2

File: recipes/pangolin/meta.yaml (hunk 1 above)
Comment (corneliusroemer) [reply recipes/pangolin/meta.yaml:25]: So we should probably update upstream setup.py/environment.yml - instead of just fixing downstream :)

Thanks for sharing your insights!

File: recipes/pangolin/meta.yaml [hunk 3]
```diff
@@ -19,23 +19,23 @@ requirements:
...
     - ucsc-fatovcf >=426
     - git
     - git-lfs
     - scorpio >=0.3.12
     - constellations >=0.0.15
     - pangolin-data >=1.2.133.2
     - conda
+    - wheel >=0.34
```
Comment (wm75) [inline recipes/pangolin/meta.yaml:38]: why is this needed?

Comment (corneliusroemer) [reply recipes/pangolin/meta.yaml:38]: It's in setup.py 🤷 
https://github.com/cov-lineages/pangolin/blob/master/setup.py

**Comment by corneliusroemer** [comment]
I tried to consolidate version requirements with pangolin upstream, where version requirements are in setup.py and environment.yml

@wm75 suggests that upstream requirements are too lenient - would be good to contribute these to the pangolin repo so that it's not confusing why bioconda is tighter than the pangolin repo specs.

File: recipes/pangolin/meta.yaml (hunk 3 above)
Comment (wm75) [reply recipes/pangolin/meta.yaml:38]: conda-forge pip brings in wheel as a dependency so there's no need for bioconda to specify it again.

**Comment by huzuner** [comment]
Can this be merged? Are there any updates on this?

**Comment by johanneskoester** [comment]
Let us merge this if possible and try the fixing of the upstream constraints afterwards. It is not uncommon to have tighter bounds here on bioconda.

File: recipes/pangolin/meta.yaml [hunk 4]
```diff
@@ -19,23 +19,23 @@
...
     - joblib >=0.11
-    - scikit-learn <1.3.0
+    - scikit-learn >=0.23.1
     - pulp >=2
     - minimap2 >=2.16
-    - snakemake-minimal >=5.13,<=7.30.1
+    - snakemake-minimal >=5.13,!=7.30.1
     - gofasta
-    - usher >=0.6.2
+    - usher >=0.5.4
```
[the remark this answers is not in the document]
Comment (johanneskoester) [reply recipes/pangolin/meta.yaml:32]: coderabbit seems right here, the issue suggests that one should use at least usher 0.6.2.

File: recipes/pangolin/meta.yaml (hunk 1 above)
Comment (johanneskoester) [reply recipes/pangolin/meta.yaml:29]:
```suggestion
    - scikit-learn >=0.23.1,<1.3.0
```

**Review by johanneskoester — APPROVED** [review]
